services/QuillLMS/app/services/vitally_integration/serialize_vitally_sales_organization.rb

```diff
@@ -0,0 +1,194 @@
+
+    {
+      diagnostics_assigned_this_year: diagnostics_assigned_this_year,
+      diagnostics_assigned_last_year: diagnostics_assigned_last_year,
+      diagnostics_completed_this_year: diagnostics_completed_this_year,
+      diagnostics_completed_last_year: diagnostics_completed_last_year,
+      percent_diagnostics_completed_this_year: percent_completed_this_year,
+      percent_diagnostics_completed_last_year: percent_completed_last_year
+    }
+  end
+
+  def subscription_rollups
+    {
+      premium_start_date: premium_start_date,
+      premium_expiry_date: premium_expiry_date,
+      district_subscription: district_subscription,
+      annual_revenue_current_contract: annual_revenue_current_contract,
+      stripe_invoice_id_current_contract: stripe_invoice_id_current_contract,
+      purchase_order_number_current_contract: purchase_order_number_current_contract
+    }
+  end
+
+  def diagnostics_assigned_between_count(start, stop)
+    @district.schools.select("array_length(classroom_units.assigned_student_ids, 1) AS assigned_students")
+      .joins(users: {
+      classrooms_i_teach: {
+        classroom_units: {
+          unit_activities: {
+           activity: :classification
+          }
+        }
+      }
+    }).where(classification: {key: ActivityClassification::DIAGNOSTIC_KEY})
+      .where(classroom_units: {created_at: start..stop})
+      .map(&:assigned_students).reject(&:blank?).sum
+  end
+
+  def diagnostics_completed_between(start, stop)
+    @district.schools.joins(users: {
+      classrooms_i_teach: {
+        classroom_units: {
+          activity_sessions: {
+            activity: :classification
+          }
+        }
+      }
+    }).where(classification: {key: ActivityClassification::DIAGNOSTIC_KEY})
+      .where(activity_sessions: {completed_at: start..stop})
+      .distinct
+      .count
+  end
+
+  def active_students(start_date=nil, end_date=nil)
+    return activities_completed_all_time.group("students.id").length if start_date.blank? && end_date.blank?
+
+    if start_date.present?
+      filtered_results = activities_completed_all_time.where("activity_sessions.completed_at >= ?", start_date).group("students.id")
+    end
+
+    if end_date.present?
+      filtered_results = activities_completed_all_time.where("activity_sessions.completed_at <= ?", end_date).group("students.id")
+    end
+
+    filtered_results.length
+  end
+
+  def activities_completed(start_date=nil, end_date=nil)
+    return activities_completed_all_time.count if start_date.blank? && end_date.blank?
+
+    if start_date.present?
+      filtered_results = activities_completed_all_time.where("activity_sessions.completed_at >= ?", start_date)
+    end
+
+    if end_date.present?
+      filtered_results = activities_completed_all_time.where("activity_sessions.completed_at <= ?", end_date)
+    end
+
+    filtered_results.count
+  end
+
+  def last_active_time
+    User.select("students.last_sign_in")
```

Comment:
this is a new method.

Reply:
Would you mind benchmarking these new queries, to minimize the possibility of introducing performance issues to production?

You can use the pattern found in `plagiarism_check_spec.rb` to tag certain specs with `benchmarking`, so that CircleCI does not run them. 

Using print statements within the `benchmark` specs, you can get run times, which you can then attach to this PR's notes. 

Alternatively, you could benchmark the entire serializer, which would mean writing fewer specs. Up to you. 

I'm happy to pair on this if that's preferable - we can find an overlap time.

Reply:
I agree with the benchmarking suggestion.  This is probably something we should be doing with all of our Vitally stuff so that we can calculate how "expensive" any new values we add are to the overnight sync process.  So while this isn't something we've done historically, it's probably a good idea to do, and this seems like an okay time to start it.

Reply:
Benchmarking specs added! Runtime reports are in the PR. @happythenewsad